src/components/VideoPlayer/Fullscreen.js

```diff
@@ -0,0 +1,90 @@
+const apiSurfaces = [
+  // Canonical non-prefixed API
+  [
+    'requestFullscreen',
+    'exitFullscreen',
+    'fullscreenElement',
+    'fullscreenEnabled',
+    'fullscreenchange',
+    'fullscreenerror'
+  ],
+  // contemporary webkit
+  [
+    'webkitRequestFullscreen',
+    'webkitExitFullscreen',
+    'webkitFullscreenElement',
+    'webkitFullscreenEnabled',
+    'webkitfullscreenchange',
+    'webkitfullscreenerror'
+  ],
+  // legacy webkit (Safari 5.1)
+  [
+    'webkitRequestFullScreen',
+    'webkitCancelFullScreen',  // That's why we check apiSurface[1] for feature detection below.
+    'webkitCurrentFullScreenElement',
+    'webkitCancelFullScreen',
+    'webkitfullscreenchange',
+    'webkitfullscreenerror'
+  ],
+  [
+    'mozRequestFullScreen',
+    'mozCancelFullScreen',
+    'mozFullScreenElement',
+    'mozFullScreenEnabled',
+    'mozfullscreenchange',
+    'mozfullscreenerror'
+  ],
+  // IE11
+  // https://msdn.microsoft.com/en-us/library/dn265028(v=vs.85).aspx
+  [
+    'msRequestFullscreen',
+    'msExitFullscreen',
+    'msFullscreenElement',
+    'msFullscreenEnabled',
+    'MSFullscreenChange',
+    'MSFullscreenError'
+  ]
+]
+
+const getFullscreenApi = () => {
+  for (let i = 0; i < apiSurfaces.length; i++) {
+    let apiSurface = apiSurfaces[i]
+    if (!!document[apiSurface[1]]) {
+      let api = {}
+      for (let j = 0; j < apiSurface.length; j++) {
+        api[apiSurfaces[0][j]] = apiSurface[j]
+      }
+      return api
+    }
+  }
+  return null
+}
+
+const Fullscreen = () => {
```

Comment:
capital letter should only be used for components (react spec) and classes. This is a class like structure but not really one. Currently this also feels more like boilerplate wrapper than actual help. You'd most likely save 20 or more lines by not abstracting it here and directly calling `document.addEventListener`, `document.removeEventListener`, `elem[api.requestFullscreen]()` and `document[api.fullscreenElement]` in the component.

An alternative would be to use a different pattern. E.g.
```js
const setupFullscreen = ({onChange, onError}) => {
  const api = getFullscreenApi()
  if (!api) {
    return
  }

  document.addEventListener(api.fullscreenchange, onChange, false)
  document.addEventListener(api.fullscreenerror, onError, false)

  return {
    request (elem) {
      const subject = elem || document.documentElement
      subject[api.requestFullscreen]()
    }
    element () {
      return document[api.fullscreenElement]
    }
    dispose () {
      document.removeEventListener(api.fullscreenchange, onChange, false)
      document.removeEventListener(api.fullscreenerror, onError, false)
    }
  }
}
```

Then just do some like this in the constructor:
```js
this.fullscreen = setupFullscreen({
  onChange () {
    this.setState({isFullscreen: this.fullscreen.element() === this.video}))
  },
  onError () {
     this.setState({isFullscreen: false})
  }
})
```
and `this.fullscreen && this.fullscreen.dispose()` in `componentWillUnmount`.

Notice the check that it is actually the video element that from the component that goes full screen and not another element. What is unclear to me: is setting `isFullscreen: false` onError truly necessary? Does the error not exclusively occur on request, before `this.fullscreen.element() === this.video` ever becomes true?